extensions/wikia/ARecoveryEngine/ARecoveryUnlockCSS.class.php

```diff
@@ -25,29 +25,45 @@ public static function getUnlockCSSUrl() {
 			if ( $cachedCriptedUrl ) {
 				return $cachedCriptedUrl;
 			} else {
-				$spQuery = self::postJson(self::API_URL . self::API_ENDPOINT, $jsonData);
+				try {
+					$spQuery = self::postJson(self::API_URL . self::API_ENDPOINT, $jsonData);
+				} catch ( Exception $e ) {
+					self::logFetchingCryptedCSSFailure( $jsonData );
+					return $wikiaCssUrl;
+				}
+
 				if ( $spQuery['code'] == 200 && self::verifyContent( $spQuery['response'] ) ) {
 					$memCache->set( $memcKey, $spQuery['response'], self::CACHE_TTL) ;
 					return $spQuery['response'];
 				} else {
-					\Wikia\Logger\WikiaLogger::instance()
-						->warning( 'Failed to fetch crypted CSS',
-							['url' => self::API_URL . self::API_ENDPOINT,
-							 'data' => $jsonData]
-						);
+					self::logFetchingCryptedCSSFailure( $jsonData );
 				}
 			}
 		}
 		return $wikiaCssUrl;
 	}
 
+	private static function logFetchingCryptedCSSFailure( $jsonData ) {
+		\Wikia\Logger\WikiaLogger::instance()
+			->warning( 'Failed to fetch crypted CSS',
+				['url' => self::API_URL . self::API_ENDPOINT,
+					'data' => $jsonData]
+			);
+	}
+
 	private static function verifyContent( $url ) {
 		$options = [
 			'returnInstance' => true,
 			'timeout' => self::TIMEOUT,
 			'noProxy' => true
 		];
-		$response = \Http::get( $url, self::TIMEOUT, $options );
+
+		try {
+			$response = \Http::get( $url, self::TIMEOUT, $options );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)